packages/button/src/Button.ts

```diff
@@ -51,6 +54,42 @@ export class Button extends SizedMixin(ButtonBase, { noDefaultSize: true }) {
         return [...super.styles, buttonStyles];
     }
 
+    protected pendingCooldown = -1;
+
+    constructor() {
+        super();
+        this.addEventListener(
+            'click',
+            (event: Event): void | boolean => {
+                if (this.pending) {
+                    event.preventDefault();
+                    event.stopImmediatePropagation();
+                    event.stopPropagation();
+                    return false;
+                }
+            },
+            {
+                capture: true,
+            }
+        );
```

Comment:
`pointer-events:none` should be better in this case so as to avoid any event propagation at all during pending state. 
Note: we are overriding the click function to avoid clicking on pending either way so even with `pointer-events:none`  we won't run into cases where users tries to click the button programatically while in pending state!